.github/workflows/autoblack.yml

```diff
@@ -0,0 +1,24 @@
+# GitHub Action that uses Black to reformat Python code (if needed) when doing a git push.
+# If all Python code in the repo is compliant with Black then this Action does nothing.
+# Otherwise, Black is run and its changes are committed to the repo.
+# https://github.com/cclauss/autoblack
+
+name: autoblack_push
+on: [push]
```

Comment:
Just a question on understanding the workflow: This is performed on **any push to Github**? Meaning that if someone pushed this to their fork on Github, as suggested by us, before submitting the PR this will already be formatted correctly when it triggers the PR Travis checks?

Looks exactly like what we needed/wanted :+1:

Reply:
> Meaning that if someone pushed this to their fork on Github, as suggested by us, before submitting the PR this will already be formatted correctly when it triggers the PR Travis checks?

If we merge this PR here, the action will be enabled for any push to a branch or fork of this repo. No user furthermore has to do anything for it.

So when someone forks, commits some changes, and opens a PR, they will see Travis starting to run tests on their latest commit, but also (if need be), the black action will add another commit to their PR for fixing the format and that commit also triggers Travis/CI.

The important part: no contributor has to be made aware or needs to know how to run black on their code. That will happen automatically.